### PR TITLE
Handle large gaps in timeseries

### PR DESCRIPTION
**Pull request by liviajakob**
Handle gaps in timeseries and a bunch of other small fixes:
- Handle cases where there is a gap in the timeseries by splitting up the timeseries at the gap within the processing chain
- Implementation to convert timeseries from Gt to mwe, so that we can now process GRACE timeseries as well
- Dealing with regions and data sources that have no data within the processing chain (ignoring it, or if we only have a trend and no annual dataset we use an auxiliary dataset for the annual variability)
- Plot fixes (wrong units displayed, label and title fixes, legend on the side of plots etc.)
- Cumulative uncertainties now calculated for cumulative timeseries, assuming that uncertainties of each timestep are independent
- minimum and maximum dates now read from config files and removed if outside period

### CODE REVIEW

File: glambie/processing/processing_helpers.py [hunk 1]
```diff
@@ -220,7 +223,8 @@ def prepare_seasonal_calibration_dataset(region_config: RegionRunConfig,
 
 def extend_annual_timeseries_if_outside_trends_period(annual_timeseries: Timeseries,
                                                       data_catalogue_trends: DataCatalogue,
-                                                      timeseries_for_extension: Timeseries) -> Timeseries:
+                                                      timeseries_for_extension: Timeseries,
+                                                      remove_trend: bool = True) -> Timeseries:
```
Comment (alex-ew) [inline glambie/processing/processing_helpers.py:227]: Missing doc string for new parameter

Comment (alex-ew) [reply glambie/processing/processing_helpers.py:227]: Also, can't see this used anywhere. It better to keep function signatures simpler when possible.

File: glambie/processing/processing_helpers.py [hunk 2]
```diff
@@ -261,3 +276,96 @@ def extend_annual_timeseries_if_outside_trends_period(annual_timeseries: Timeser
...
+        previous_index = split_index + 1
+    # plus append last / full split in the end
+    split_timeseries_dataframes.append(df_timeseries.iloc[previous_index:].reset_index(drop=True))
+    return split_timeseries_dataframes
+
+
+def set_unneeded_columns_to_nan(data_catalogue: DataCatalogue) -> DataCatalogue:
+    """
+    Sets data columns of TimeseriesData not needed in algorithm to NaN within a DataCatalogue
+    to simplify object manipulating
```
Comment (alex-ew) [inline glambie/processing/processing_helpers.py:353]:
```suggestion
    to simplify object manipulation
```

File: glambie/processing/processing_helpers.py [hunk 3]
```diff
@@ -261,3 +276,96 @@ def extend_annual_timeseries_if_outside_trends_period(annual_timeseries: Timeser
...
+        split_timeseries_dataframes.append(df_timeseries.iloc[previous_index:split_index + 1].reset_index(drop=True))
+        previous_index = split_index + 1
+    # plus append last / full split in the end
+    split_timeseries_dataframes.append(df_timeseries.iloc[previous_index:].reset_index(drop=True))
+    return split_timeseries_dataframes
+
+
+def set_unneeded_columns_to_nan(data_catalogue: DataCatalogue) -> DataCatalogue:
+    """
+    Sets data columns of TimeseriesData not needed in algorithm to NaN within a DataCatalogue
```
Comment (alex-ew) [inline glambie/processing/processing_helpers.py:352]: Are these fields not needed by any algorithm? Or this applies to specific algorithms? It appears to be specific to a particular algorithm but it's unclear which one.

**Review by alex-ew — CHANGES_REQUESTED** [review]
Very clear, readable unit tests! However, this should probably have been ~3 smaller PRs. I've left some minor comments/suggestions but nothing major.

File: glambie/plot/plot_helpers.py [hunk 4]
```diff
@@ -226,7 +226,8 @@ def finalise_save_to_file_and_close_plot(output_filepath: str):
 def add_labels_axlines_and_title(axes: mpl.pyplot.axes,
                                  unit: str,
                                  title: str,
-                                 legend_fontsize: int = 9):
+                                 legend_fontsize: int = 9,
+                                 legend_outside_plot: bool = True):
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/plot/plot_helpers.py:230]: I will use it:)

File: glambie/processing/processing_helpers.py (hunk 1 above)
Comment (liviajakob) [reply glambie/processing/processing_helpers.py:227]: yes, actually this is just a "might use that in the far future". Will remove the param!

File: glambie/processing/processing_helpers.py (hunk 3 above)
Comment (liviajakob) [reply glambie/processing/processing_helpers.py:352]: Yes, so basically they are not needed in the glambie processing algo, however we will use them for other bits of analysis and custom plotting of the datasets.

I clarified.

File: glambie/processing/processing_helpers.py (hunk 2 above)
Comment (liviajakob) [reply glambie/processing/processing_helpers.py:353]: ooft you're so good at spotting random typos and errors in the docstrings! done:)

File: glambie/processing/process_global_results.py [hunk 5]
```diff
@@ -42,8 +42,17 @@ def run_global_results(glambie_run_config: GlambieRunConfig,
     regional_results_catalogue_homogenized = _homogenize_regional_results_to_calendar_year(glambie_run_config,
                                                                                            regional_results_catalogue,
                                                                                            original_data_catalogue)
-    global_timeseries = _combine_regional_results_into_global(regional_results_catalogue_homogenized)
+    # remove any dates outside config timeperiod:
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/processing/process_global_results.py:45]: done

File: glambie/processing/process_global_results.py [hunk 6]
```diff
@@ -42,8 +42,17 @@ def run_global_results(glambie_run_config: GlambieRunConfig,
...
+    for dataset in regional_results_catalogue_homogenized.datasets:
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/processing/process_global_results.py:46]: I have made it a method in Timeseries (reduce_to_date_window()) for each dataset and then a helper function in processing_helpers

File: glambie/plot/plot_helpers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (alex-ew) [reply glambie/plot/plot_helpers.py:230]: Ok. I'll let this one slide as it's minor, but it's better practice in general to implement changes only when you need them. Too frequently people will add complexity that's never required.

File: glambie/processing/process_global_results.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (alex-ew) [reply glambie/processing/process_global_results.py]: Much cleaner

**Review by alex-ew — APPROVED** [review]
Thanks for the changes

**Pull request merged**